### PR TITLE
type combinator: fix integer ranges operations

### PR DESCRIPTION
**Pull request by hrach**
Another minute, another PR. 💦

### CODE REVIEW

File: tests/PHPStan/Analyser/data/isset.php [hunk 1]
```diff
@@ -10,16 +10,16 @@ class Foo
 	 */
 	public function doFoo(array $integers, string $string, $mixedIsset, $mixedArrayKeyExists)
 	{
-		if (rand(0, 1) === 0) {
+		if (rand(0, 10) === 0) {
```
Comment (hrach) [inline tests/PHPStan/Analyser/data/isset.php]: It seems that PhpStan already has something definition of pure functions and therefore using rand with same args is cached. 🤔 But not sure what is the current state.

Comment (ondrejmirtes) [reply tests/PHPStan/Analyser/data/isset.php]: Actually not yet, it currently dumbly caches `rand(0, 1)` result for all functions.

**Comment by ondrejmirtes** [comment]
I really don't understand the math, @JanTvrdik has to review this. Thank you.

**Comment by ondrejmirtes** [comment]
OK, let's just see if anyone finds a bug in this :) Thank you very much!

**Pull request merged**